testing/testcharm.go

```diff
@@ -61,14 +63,29 @@ type CharmSpec struct {
 	Revision int
 }
 
-// NewCharm returns a new charm
+type file struct {
+	path string
+	data []byte
+	perm os.FileMode
+}
+
+// NewCharm returns a charm following the given specification.
 func NewCharm(c *gc.C, spec CharmSpec) *Charm {
+	return newCharm(spec)
```

Comment:
For compatibility should this catch paniccy panics and replace them with asserty panics?